test/byContract/IexecPoco/02_contribute.test.ts

```diff
@@ -0,0 +1,517 @@
+                    taskId,
+                    worker.resultDigest,
+                    worker.wallet,
+                );
+                await iexecWrapper.depositInIexecAccount(worker.wallet, workerTaskStake);
+                const viewFrozenOf = (address: string) =>
+                    iexecPoco.frozenOf(address).then((frozen) => frozen.toNumber());
+                const frozenBefore = await viewFrozenOf(workerAddress);
+                contributeBlockTimestamp = await setNextBlockTimestamp();
+                const tx = await iexecPoco
+                    .connect(worker.wallet)
+                    .contribute(
+                        taskId,
+                        resultHash,
+                        resultSeal,
+                        enclave.address,
+                        await buildAndSignPocoClassicEnclaveMessage(
+                            resultHash,
+                            resultSeal,
+                            enclave,
+                        ),
+                        await buildAndSignContributionAuthorizationMessage(
+                            workerAddress,
+                            taskId,
+                            enclave.address,
+                            sms,
+                        ),
+                    );
+                const contribution = await iexecPoco.viewContribution(taskId, workerAddress);
+                expect(contribution.status).equal(ContributionStatusEnum.CONTRIBUTED);
+                expect(contribution.resultHash).equal(resultHash);
+                expect(contribution.resultSeal).equal(resultSeal);
+                expect(contribution.enclaveChallenge).equal(enclave.address);
+                task = await iexecPoco.viewTask(taskId);
+                expect(task.contributors.length).equal(i + 1);
+                expect(task.contributors[i]).equal(workerAddress);
+                await expect(tx)
+                    .to.changeTokenBalances(
+                        iexecPoco,
+                        [workerAddress, proxyAddress],
+                        [-workerTaskStake, workerTaskStake],
+                    )
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(workerAddress, proxyAddress, workerTaskStake);
+                expect(await viewFrozenOf(workerAddress)).equal(frozenBefore + workerTaskStake);
+                await expect(tx)
+                    .to.emit(iexecPoco, 'Lock')
+                    .withArgs(workerAddress, workerTaskStake)
+                    .to.emit(iexecPoco, 'TaskContribute')
+                    .withArgs(taskId, workerAddress, resultHash);
+                expect(contribution.weight).equal(1); // no worker history
+                // m_consensus does not have any getter for checking group & total
+                const taskConsensusEvent: [contract: any, eventName: string] = [
+                    iexecPoco,
+                    'TaskConsensus',
+                ];
```

Comment:
Declare it inside the next `if`.

Reply:
I don't understand you :ear:

Reply:
I didn't notice that `taskConsensusEvent` is used in both if and else. Never mind.